### PR TITLE
[Health check] Add logs with the details about the check

### PR DESCRIPTION
**Pull request by Desvelao**
### Description
This PR adds logs with the details about the checks.
- Added button to open/close the check details
- Added a new section in Settings: `Miscellaneous`:
  - Health check: run in `debug` mode
- Refactorized how the check validator function is managed
- Refactorized another function
- Some adjustments in Settings section related to spaces

### Checks
- Health check:
  - Open/close the check details with the button added after the check result
  - When a check does an `action`, the button to open the details should have a notification icon
  
- Settings > Miscellaneous:
  - The new section should be visible and accessible in the Settings section
    - Try the link in the main menu
    - Try the tab in Settings
  - Run the Health check in `debug` mode:
    - Should redirect to health check
    - A `Continue` button should appear when all checks are ready
    
- Considers reviewing the messages displayed in the logs
- Try with index patterns not created

### Suggested improvements
- [x] Remove the notification after the check logs are opened

Closes: #3208

### CODE REVIEW

**Review by CPAlejandro — CHANGES_REQUESTED** [review]
- When stopping the manager service and executing the health-check, an error of an uncontrolled undefined appears.
image

- As a suggestion, maybe when a red dot appears and you open the logs, it could disappear?

**Comment by Desvelao** [comment]
I like the suggestion to remove the notification after opening the check log. I will wait until somebody reviews the code and tests it

**Comment by Desvelao** [comment]
> * When stopping the manager service and executing the health-check, an error of an uncontrolled undefined appears.
>   image

I tried to replicate the problem and I can't get the same error displayed in the screenshot. Could someone try to replicate?

I tried:
- Stop the `wazuh-manager` service
```service wazuh-manager stop```
- Turn off the machine with the manager (in my environment using Docker container, I stoped the container)

**Review by CPAlejandro — COMMENTED** [review]
- Could we control the use of toast to not be able to display the same one twice, which can be annoying?
image (3)
- When moving from branch 4.2-7.10 to this branch and executing the health-check from / settings / Miscellaneous I get the following error
x-special/nautilus-clipboard
copy
image (4)
image (5)

**Comment by Desvelao** [comment]
### Changes
- Fix unknown error after restarting the cluster
- Removed the notification dot when opening for the first time the check details
- Updated tests

**Review by CPAlejandro — COMMENTED** [review]
Testing: LGTM!

**Review by pablomarga — APPROVED** [review]
CR: :heavy_check_mark: 
Testing: :heavy_check_mark:

**Review by CPAlejandro — APPROVED** [review]

**Comment by Desvelao** [comment]
Someone to review the log messages @wazuh/frontend .

**Comment by asteriscos** [comment]
Team,
made some small improvements to the UI: 

- Fixed some verbose log texts
- Removed redundant status texts
- Improved the styling

A quick review is needed please!

healthcheck

File: public/components/health-check/components/check-result.tsx [hunk 1]
```diff
@@ -112,21 +150,73 @@ export function CheckResult(props) {
...
+    setVerboseIsOpen(state => !state);
+  },[checkDidSomeAction]);
+
+  const inspectLogsButton = isCheckStarted && haveLogs ? 
+    <EuiToolTip
+      position='top'
+      content={tooltipVerboseButton}
+      >
+      <EuiButtonEmpty size='xs' onClick={switchVerboseDetails} isDisabled={!haveLogs} textProps={{ style:{overflow: 'visible' , position: 'relative'}}}>
+        <EuiButtonIcon
```
Comment (asteriscos) [inline public/components/health-check/components/check-result.tsx:184]: This button inception could be avoided?

File: public/components/health-check/components/check-result.tsx [hunk 2]
```diff
@@ -112,21 +150,73 @@ export function CheckResult(props) {
...
+  const haveLogs = verboseInfo.length > 0;
+  
+  const switchVerboseDetails = useCallback(() => {
+    if(checkDidSomeAction && !verboseInfoWithNotificationWasOpened.current){
+      verboseInfoWithNotificationWasOpened.current = true;
+    };
+    setVerboseIsOpen(state => !state);
+  },[checkDidSomeAction]);
+
+  const inspectLogsButton = isCheckStarted && haveLogs ?
```
Comment (frankeros) [inline public/components/health-check/components/check-result.tsx:178]: pls move the button and logs component to separate child components in order to keep a short class

**Review by frankeros — CHANGES_REQUESTED** [review]
Test: :heavy_check_mark: 
CR: changes requestd

I tested and it looks good. But I think that is a little confusing that the user gets the chance to open the logs then the health-check redirects automatically, 
I would like that the button and logs to be displayed only when there is an error in normal mode. 
In verbose mode it's ok.

File: public/components/health-check/components/check-result.tsx (hunk 2 above)
Comment (asteriscos) [reply public/components/health-check/components/check-result.tsx]: Great idea, already split the components. Please review them!

**Comment by asteriscos** [comment]
> Test: 
> CR: changes requestd
> 
> I tested and it looks good. But I think that is a little confusing that the user gets the chance to open the logs then the health-check redirects automatically,
> I would like that the button and logs to be displayed only when there is an error in normal mode.
> In verbose mode it's ok.

Sounds good. Now the log buttons will show when there's an error or in verbose mode. Please review it!

**Review by CPAlejandro — APPROVED** [review]
Testing: LGTM!

**Review by frankeros — APPROVED** [review]
Test: :heavy_check_mark: 
CR: :heavy_check_mark: 
Amazing job guys!! :tada:

**Pull request merged**